vector-im/element-android
# Fix animated webp playback

## Type of change


## Content



Allowed Animated WebP to play correctly.

This involves making Glide transformations optional in order to prevent mimetypes that Glide doesn't support from throwing exceptions. This has a very minor impact - making rounded corners not work for animated WebP files in the timeline. However, I see this as an improvement over it not working at all.

## Motivation and context



Animated WebP files are delivered by default from the Facebook Bridge (https://github.com/mautrix/facebook) so this has been a pain point for me.

Element Android issues:
https://github.com/vector-im/element-android/issues/2695
https://github.com/vector-im/element-android/issues/4199

Animated WebP is supported on android versions natively. The only issue is that the Glide library doesn't support it outside of rendering:
https://github.com/bumptech/glide/issues/571

This could alternatively potentially be fixed by using https://github.com/zjupure/GlideWebpDecoder
However, this feels like a change that could potentially introduce more issues, as that library only supports a couple of recent versions of Glide, which would necessitate keeping those in-line. I'm not sure how well maintained the library is, and this could be worse than just accepting the minor differences in missing rounded corners for animated WebP files.

## Screenshots / GIFs


|Before|After|
|-|-|
|![animated-webp-before](https://user-images.githubusercontent.com/7509232/218303270-8f213e18-2fb4-4f13-bef3-c9cf3d35b866.gif)|![animated-webp-after](https://user-images.githubusercontent.com/7509232/218303290-c73cbe32-a487-49c3-9650-a346d9080b25.gif)|

There is a minor annoyance introduced by this - non-animated webp files will have a "play" button on them. This is unavoidable without some major changes to the image viewer. We'd need to pull the image and read the VP8X header for the ANIM bit to see if it's an animated webp. The same issue is already present with non-animated Gifs. 

## Tests



- Run in Android Studio
- Open a room with animated WebP files
- Tap into the animated WebP, ensure it still plays
- Tap into an animated gif, ensure it still plays
- Tap into a still WebP, ensure there are no changes (due to now being marked as `playable`)

## Tested devices

- OS version(s): Pixel 6 API 21 on Emulator w/ Lollipop, Pixel 4A API 31 on Emulator, Nothing Phone One (physical) w/ Android 12

The changes don't do anything particularly useful on older versions of Android - but it's not more broken than it was. Android requires API v28 to play Animated WebP by the looks of it. At the moment, it would just be still images, whereas with this change they can at least see that they're supposed to be animated. 

## Checklist
